Okay, I've updated the PR with the suggestions by changing the original commits to keep the history clean. Diff can be seen here: https://github.com/adafruit/Adafruit_Board_Toolkit/compare/eecfa51e7ef18bcc346b13de44ae370c4fe24e6d..76bd5a1d7cffef8b2a16cc75c66cf678b837f24c

However now one of the pre-commit hooks is failing:
https://github.com/adafruit/Adafruit_Board_Toolkit/actions/runs/3688091213/jobs/6242484092
```
[INFO] Installing environment for https://github.com/pycqa/pylint.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
black....................................................................Failed
- hook id: black
- exit code: 1

Traceback (most recent call last):
  File "/home/runner/.cache/pre-commit/repo7b33cyhu/py_env-python3/bin/black", line 8, in <module>
    sys.exit(patched_main())
             ^^^^^^^^^^^^^^
  File "/home/runner/.cache/pre-commit/repo7b33cyhu/py_env-python3/lib/python3.11/site-packages/black/__init__.py", line 6606, in patched_main
    patch_click()
  File "/home/runner/.cache/pre-commit/repo7b33cyhu/py_env-python3/lib/python3.11/site-packages/black/__init__.py", line 6595, in patch_click
    from click import _unicodefun  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ImportError: cannot import name '_unicodefun' from 'click' (/home/runner/.cache/pre-commit/repo7b33cyhu/py_env-python3/lib/python3.11/site-packages/click/__init__.py)

reuse....................................................................Passed
Check Yaml...............................................................Passed
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
pylint (library code)....................................................Failed
- hook id: pylint
- exit code: 1

Traceback (most recent call last):
  File "/home/runner/.cache/pre-commit/repod89re8iu/py_env-python3.11/bin/pylint", line 8, in <module>
    sys.exit(run_pylint())
             ^^^^^^^^^^^^
  File "/home/runner/.cache/pre-commit/repod89re8iu/py_env-python3.11/lib/python3.11/site-packages/pylint/__init__.py", line 19, in run_pylint
    from pylint.lint import Run as PylintRun
  File "/home/runner/.cache/pre-commit/repod89re8iu/py_env-python3.11/lib/python3.11/site-packages/pylint/lint/__init__.py", line 75, in <module>
    from pylint.lint.parallel import check_parallel
  File "/home/runner/.cache/pre-commit/repod89re8iu/py_env-python3.11/lib/python3.11/site-packages/pylint/lint/parallel.py", line 7, in <module>
    from pylint import reporters
  File "/home/runner/.cache/pre-commit/repod89re8iu/py_env-python3.11/lib/python3.11/site-packages/pylint/reporters/__init__.py", line 24, in <module>
    from pylint import utils
  File "/home/runner/.cache/pre-commit/repod89re8iu/py_env-python3.11/lib/python3.11/site-packages/pylint/utils/__init__.py", line 46, in <module>
    from pylint.utils.ast_walker import ASTWalker
  File "/home/runner/.cache/pre-commit/repod89re8iu/py_env-python3.11/lib/python3.11/site-packages/pylint/utils/ast_walker.py", line 6, in <module>
    from astroid import nodes
  File "/home/runner/.cache/pre-commit/repod89re8iu/py_env-python3.11/lib/python3.11/site-packages/astroid/__init__.py", line 45, in <module>
    import wrapt
  File "/home/runner/.cache/pre-commit/repod89re8iu/py_env-python3.11/lib/python3.11/site-packages/wrapt/__init__.py", line 10, in <module>
    from .decorators import (adapter_factory, AdapterFactory, decorator,
  File "/home/runner/.cache/pre-commit/repod89re8iu/py_env-python3.11/lib/python3.11/site-packages/wrapt/decorators.py", line 34, in <module>
    from inspect import ismethod, isclass, formatargspec
ImportError: cannot import name 'formatargspec' from 'inspect' (/opt/hostedtoolcache/Python/3.11.0/x64/lib/python3.11/inspect.py)

pylint (examples code)...................................................Passed
Error: Process completed with exit code 1.

```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -22,13 +22,15 @@ jobs:
         awk -F '\/' '{ print tolower($2) }' |
         tr '_' '-'
         )
-    - name: Set up Python 3.6
-      uses: actions/setup-python@v1
+    - name: Set up Python
+      uses: actions/setup-python@v4
       with:
-        python-version: 3.6
+        python-version: '3.x'
     - name: Versions
       run: |
-        python3 --version
+        python --version
+        python -m pip --version
+        python -m pip list
     - name: Checkout Current Repo
       uses: actions/checkout@v1
       with:

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -19,7 +19,7 @@ jobs:
         echo ::set-output name=setup-py::$( find . -wholename './setup.py' )
     - name: Set up Python
       if: contains(steps.need-pypi.outputs.setup-py, 'setup.py')
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v4
       with:
         python-version: '3.x'
     - name: Install dependencies

--- a/.gitignore
+++ b/.gitignore
@@ -17,3 +17,4 @@ bundles
 dist
 **/*.egg-info
 .vscode
+.venv/*

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -4,7 +4,7 @@
 
 repos:
 -   repo: https://github.com/python/black
-    rev: 20.8b1
+    rev: 22.12.0
     hooks:
     - id: black
       exclude: _list_ports_.*.py
@@ -19,7 +19,7 @@ repos:
     -   id: end-of-file-fixer
     -   id: trailing-whitespace
 -   repo: https://github.com/pycqa/pylint
-    rev: pylint-2.7.1
+    rev: v2.15.8
     hooks:
     -   id: pylint
         name: pylint (library code)

--- a/docs/conf.py
+++ b/docs/conf.py
@@ -63,7 +63,7 @@ release = "1.0"
 #
 # This is also used if you do content translation via gettext catalogs.
 # Usually you set "language" from the command line for these cases.
-language = None
+language = "en"
 
 # List of patterns, relative to source directory, that match files and
 # directories to ignore when looking for source files.

